dd-java-agent/agent-iast/src/main/java/com/datadog/iast/model/VulnerabilityType.java

```diff
@@ -1,34 +1,55 @@
-  VulnerabilityType WEAK_HASH = new VulnerabilityTypeImpl(VulnerabilityTypes.WEAK_HASH);
-  VulnerabilityType INSECURE_COOKIE = new VulnerabilityTypeImpl(VulnerabilityTypes.INSECURE_COOKIE);
+  VulnerabilityType WEAK_CIPHER =
+      new VulnerabilityTypeImpl(VulnerabilityTypes.WEAK_CIPHER, NOT_MARKED);
+  VulnerabilityType WEAK_HASH = new VulnerabilityTypeImpl(VulnerabilityTypes.WEAK_HASH, NOT_MARKED);
+  VulnerabilityType INSECURE_COOKIE =
+      new VulnerabilityTypeImpl(VulnerabilityTypes.INSECURE_COOKIE, NOT_MARKED);
   VulnerabilityType NO_HTTPONLY_COOKIE =
-      new VulnerabilityTypeImpl(VulnerabilityTypes.NO_HTTPONLY_COOKIE);
+      new VulnerabilityTypeImpl(VulnerabilityTypes.NO_HTTPONLY_COOKIE, NOT_MARKED);
   VulnerabilityType NO_SAMESITE_COOKIE =
-      new VulnerabilityTypeImpl(VulnerabilityTypes.NO_SAMESITE_COOKIE);
+      new VulnerabilityTypeImpl(VulnerabilityTypes.NO_SAMESITE_COOKIE, NOT_MARKED);
 
-  InjectionType SQL_INJECTION = new InjectionTypeImpl(VulnerabilityTypes.SQL_INJECTION, ' ');
+  InjectionType SQL_INJECTION =
+      new InjectionTypeImpl(
+          VulnerabilityTypes.SQL_INJECTION, VulnerabilityMarks.SQL_INJECTION_MARK, ' ');
   InjectionType COMMAND_INJECTION =
-      new InjectionTypeImpl(VulnerabilityTypes.COMMAND_INJECTION, ' ');
+      new InjectionTypeImpl(
+          VulnerabilityTypes.COMMAND_INJECTION, VulnerabilityMarks.COMMAND_INJECTION_MARK, ' ');
   InjectionType PATH_TRAVERSAL =
-      new InjectionTypeImpl(VulnerabilityTypes.PATH_TRAVERSAL, File.separatorChar);
-  InjectionType LDAP_INJECTION = new InjectionTypeImpl(VulnerabilityTypes.LDAP_INJECTION, ' ');
-  InjectionType SSRF = new InjectionTypeImpl(VulnerabilityTypes.SSRF, ' ');
+      new InjectionTypeImpl(
+          VulnerabilityTypes.PATH_TRAVERSAL,
+          VulnerabilityMarks.PATH_TRAVERSAL_MARK,
+          File.separatorChar);
+  InjectionType LDAP_INJECTION =
+      new InjectionTypeImpl(
+          VulnerabilityTypes.LDAP_INJECTION, VulnerabilityMarks.LDAP_INJECTION_MARK, ' ');
+  InjectionType SSRF =
+      new InjectionTypeImpl(VulnerabilityTypes.SSRF, VulnerabilityMarks.SSRF_MARK, ' ');
   InjectionType UNVALIDATED_REDIRECT =
-      new InjectionTypeImpl(VulnerabilityTypes.UNVALIDATED_REDIRECT, ' ');
-  VulnerabilityType WEAK_RANDOMNESS = new VulnerabilityTypeImpl(VulnerabilityTypes.WEAK_RANDOMNESS);
+      new InjectionTypeImpl(
+          VulnerabilityTypes.UNVALIDATED_REDIRECT,
+          VulnerabilityMarks.UNVALIDATED_REDIRECT_MARK,
+          ' ');
+  VulnerabilityType WEAK_RANDOMNESS =
+      new VulnerabilityTypeImpl(VulnerabilityTypes.WEAK_RANDOMNESS, NOT_MARKED);
 
-  InjectionType XPATH_INJECTION = new InjectionTypeImpl(VulnerabilityTypes.XPATH_INJECTION, ' ');
+  InjectionType XPATH_INJECTION =
+      new InjectionTypeImpl(
+          VulnerabilityTypes.XPATH_INJECTION, VulnerabilityMarks.XPATH_INJECTION_MARK, ' ');
 
   String name();
 
+  int mark();
```

Comment:
```suggestion
  /** A bit flag to ignore tainted ranges for this vulnerability. Set to 0 if none. */
  int mark();
```

Or something like this,

Reply:
done